zerver/lib/email_mirror.py

```diff
@@ -171,11 +171,19 @@ class ZulipEmailForwardUserError(ZulipEmailForwardError):
     pass
 
 def send_zulip(sender: UserProfile, stream: Stream, topic: str, content: str) -> None:
+    trunc_topic = truncate_topic(topic)
+    if len(trunc_topic) < len(topic):
+        if not (Message.objects.filter(
+                subject = trunc_topic,
```

Comment:
This should use the helpers from `zerver/lib/topic.py` rather than accessing `topic` directly.

Reply:
Sorry, I am not really sure what that means. Are you referring to `TOPIC_NAME` ?
Actually, I tried to do something similar to a query in an existing method in `models.py` , where they use subject

```
def get_context_for_message(message: Message) -> Sequence[Message]:
    # TODO: Change return type to QuerySet[Message]
    return Message.objects.filter(
        recipient_id=message.recipient_id,
        subject=message.subject,
        id__lt=message.id,
        date_sent__gt=message.date_sent - timedelta(minutes=15),
    ).order_by('-id')[:10]
```

Reply:
@madhavsikka We want to avoid referring to ``subject`` in the sense of message topic - you can use ``filter_by_topic_name_via_message`` from topic.py instead of the ``subject=`` part of the filter. It makes the filtering case-insensitive but I think that's actually desirable :thinking:

Reply:
@madhavsikka We want to avoid referring to ``subject`` in the sense of message topic - you can use ``messages_for_topic`` from topic.py instead of the ``subject=`` part of the filter. It makes the filtering case-insensitive but I think that's actually desirable :thinking:

Reply:
Could you check now, if this is what's required?

Reply:
Yes, that should be good :slightly_smiling_face: